### PR TITLE
Show inject usage when args are passed incorrectly

### LINKED ISSUE #5384: bundle inject must show the usage when it is used incorrectly
$ bundle inject 'pry' 1 '>0'
--- ERROR REPORT TEMPLATE -------------------------------------------------------
# Error Report

## Questions

Please fill out answers to these questions, it'll help us figure out
why things are going wrong.

- **What did you do?**

  I ran the command `bundle inject 'pry' 1 '>0'`

- **What did you expect to happen?**

  I expected Bundler to show error message saying the correct format and usage of bundle inject.

- **What happened instead?**

  Showing to open an issue

- **Have you tried any solutions posted on similar issues in our issue tracker, stack overflow, or google?**
 
 Asked in slack channel. 

- **Have you read our issues document, https://github.com/bundler/bundler/blob/master/ISSUES.md?**

  yes

## Backtrace

```
NoMethodError: undefined method `start_with?' for nil:NilClass
  /Users/bundler/lib/bundler/cli/inject.rb:23:in `block (2 levels) in run'
  /Users/bundler/lib/bundler/cli/inject.rb:23:in `any?'
  /Users/bundler/lib/bundler/cli/inject.rb:23:in `block in run'
  /Users/bundler/lib/bundler/cli/inject.rb:21:in `each_slice'
  /Users/bundler/lib/bundler/cli/inject.rb:21:in `run'
  /Users/bundler/lib/bundler/cli.rb:468:in `inject'
  /Users/bundler/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
  /Users/bundler/lib/bundler/vendor/thor/lib/thor/invocation.rb:126:in `invoke_command'
  /Users/bundler/lib/bundler/vendor/thor/lib/thor.rb:359:in `dispatch'
  /Users/bundler/lib/bundler/cli.rb:21:in `dispatch'
  /Users/bundler/lib/bundler/vendor/thor/lib/thor/base.rb:440:in `start'
  /Users/bundler/lib/bundler/cli.rb:11:in `start'
  /Users/bundler/exe/bundle:32:in `block in <main>'
  /Usersbundler/lib/bundler/friendly_errors.rb:121:in `with_friendly_errors'
  /Users/bundler/exe/bundle:24:in `<main>'
```

## Environment

```
Bundler   1.14.3
Rubygems  2.5.1
Ruby      2.3.1p112 (2016-04-26 revision 54768) [x86_64-darwin15]
GEM_HOME  /Users/.rvm/gems/ruby-2.3.1
GEM_PATH  /Users/.rvm/gems/ruby-2.3.1:/Users/.rvm/gems/ruby-2.3.1@global
RVM       1.28.0 (latest)
Git       2.10.1
Platform  x86_64-darwin-15
OpenSSL   OpenSSL 1.0.2j  26 Sep 2016
rubygems-bundler (1.4.4)
```

## Bundler settings

```
gem.test
  Set for the current user (/Users/.bundle/config): "false"
gem.mit
  Set for the current user (/Users/.bundle/config): "true"
gem.coc
  Set for the current user (/Users/.bundle/config): "true"
trampoline_disable
  Set via BUNDLE_TRAMPOLINE_DISABLE: "1"
```

### PR DESCRIPTION
**Pull request by Shekharrajak**
Closes #5384

Now it shows the error : 

```
$ bundle inject "gem_name" 1 "v"
ERROR: "bundle inject" was called with arguments ["gem_name", "1", "v"]
Usage: "bundle inject GEM VERSION"
$ bundle inject "gem_name" "v" 1
ERROR: "bundle inject" was called with arguments ["gem_name", "v", "1"]
Usage: "bundle inject GEM VERSION"

```

### CODE REVIEW

**Comment by Shekharrajak** [comment]
I didn't understand this error: https://travis-ci.org/bundler/bundler/jobs/197244354. Can any one help ?

**Comment by hmistry** [comment]
@Shekharrajak I like this idea. I recommend changing the error statement to something that reaffirms the error instead of only depending on the "ERROR" word in the beginning. e.g.
```
$ bundle inject "gem_name" 1 "v"
ERROR: "bundle inject" was incorrectly called with arguments ["gem_name", "1", "v"]
Usage: "bundle inject GEM VERSION"
```

@segiddins I'm fuzzy on this but does inject not support adding multiple gems? I ask because some of that portion of the code is being removed with this commit.

**Comment by segiddins** [comment]
@hmistry I have no idea

**Comment by indirect** [comment]
@Shekharrajak thanks! this is looking good. could you add a test that calls inject with invalid arguments and checks for the correct error message?

**Comment by Shekharrajak** [comment]
Thanks @segiddins , @hmistry, @indirect  for review. Yes, right now it must take one gem. I don't know how to use this [option](https://github.com/bundler/bundler/pull/5389/files#diff-9fc4de3771c95e6c1bbbeba954b5a064R9) variable to get `source`, `group`. If we find a way to do that then using `inject` we can write `source`, `group` also.(And may be using good implementation of `bundle add` we can add multiple gems, using`inject` method internally).

File: spec/commands/inject_spec.rb [hunk 1]
```diff
@@ -42,6 +42,14 @@
     end
   end
 
+    it "fails when more than 2 arguments are passed" do
```
Comment (segiddins) [inline spec/commands/inject_spec.rb]: indentation

**Comment by Shekharrajak** [comment]
>  I recommend changing the error statement to something that reaffirms the error instead of only depending on the "ERROR" word in the beginning. e.g.

@hmistry , I think 1st line of error message is same for all other commands too. So I am not changing it.

File: spec/commands/inject_spec.rb (hunk 1 above)
Comment (Shekharrajak) [inline spec/commands/inject_spec.rb]: Thanks I have resolved the problem.

**Comment by hmistry** [comment]
@Shekharrajak I see, you're just keeping the old error message that was there. I thought you were modifying it and as such felt it was unclear as to what a user did wrong. It says you called the option with some arguments and as a user I'd go "yeah, so what? why did you error?" grumble, grumble, grumble. A better message would tell me exactly what I did wrong and suggest how to fix it. We'd like to reduce such issues from users down the road.

It's entirely your choice to improve it or not as it is your time. If you choose not to, I'll note it for future fixes. Thank you so much for taking the time to do the above commit. 😄

**Comment by segiddins** [comment]
@bundlerbot r+

**Pull request merged**